src/java/htsjdk/samtools/SAMSequenceDictionary.java

```diff
@@ -163,8 +176,124 @@ public boolean equals(Object o) {
+                    + " doesn't exist in dictionary.");
+        // same name, nothing to do
+        if (originalName.equals(altName))
+            return originalSeqRecord;
+        final SAMSequenceRecord altSeqRecord = getSequence(altName);
+        if (altSeqRecord != null) {
+            // alias was already set to the same record
+            if (altSeqRecord.equals(originalSeqRecord))
+                return originalSeqRecord;
+            // alias was already set to another record
+            throw new IllegalArgumentException("Alias " + altName
+                    + " was already set to " + altSeqRecord.getSequenceName());
+        }
+        mSequenceMap.put(altName, originalSeqRecord);
+        return originalSeqRecord;
+    }
+
+    /**
+     * return a MD5 sum for ths dictionary, the checksum is re-computed each
+     * time this method is called.
+     * 
+     * <pre>
+     * md5( (seq1.md5_if_available) + ' '+(seq2.name+seq2.length) + ' '+...)
+     * </pre>
+     * 
+     * @return a MD5 checksum for this dictionary or the empty string if it is
+     *         empty
+     */
+    public String md5() {
+        if (isEmpty())
+            return "";
+        try {
+            final MessageDigest md5 = MessageDigest.getInstance("MD5");
+            md5.reset();
+            for (final SAMSequenceRecord ssr : mSequences) {
+                if (ssr.getSequenceIndex() > 0)
+                    md5.update((byte) ' ');
+                final String md5_tag = ssr.getAttribute(SAMSequenceRecord.MD5_TAG);
+                if (md5_tag != null) {
+                    md5.update(md5_tag.getBytes());
+                } else {
+                    md5.update(ssr.getSequenceName().getBytes());
+                    md5.update(String.valueOf(ssr.getSequenceLength()).getBytes());
+                }
+            }
+            String hash = new BigInteger(1, md5.digest()).toString(16);
+            if (hash.length() != 32) {
+                final String zeros = "00000000000000000000000000000000";
+                hash = zeros.substring(0, 32 - hash.length()) + hash;
+            }
+            return hash;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     @Override
     public int hashCode() {
         return mSequences.hashCode();
     }
-}
\ No newline at end of file
+
+    @Override
+    public String toString() {
+        return "SAMSequenceDictionary:( sequences:"+ size()+
+                " length:"+ getReferenceLength()+" "+
+                " md5:"+md5()+")";
+    }
+
+    /**
+     * @param sequenceIndex
+     *            contig id
+     * @param position
+     *            position in the contig
+     * @returns the number of bases seen from the first sequence at position 0
+     *          to the sequenceIndex/position or -1 on error
+     */
+    public long getTotalBases(final int sequenceIndex, final int position) {
```

Comment:
Is it really appropriate to put utility functions like this here?  Could folks not compute this themselves?  I don't want to dump a ton of utility functions into these library classes.  Can you make a case for its usefulness for others?